portal/migrations/versions/55469bdd181f_.py

```diff
@@ -0,0 +1,111 @@
+"""Correct QNR qb_id when assigned to the incorrect research protocol
+
+Revision ID: 55469bdd181f
+Revises: 9026af5fe040
+Create Date: 2019-02-04 13:36:04.709868
+
+"""
+from portal.database import db
+from portal.models.qb_status import QB_Status
+from portal.models.qb_timeline import QBT, invalidate_users_QBT
+from portal.models.questionnaire_bank import QuestionnaireBank
+from portal.models.questionnaire_response import QuestionnaireResponse
+from portal.models.research_protocol import ResearchProtocol
+from portal.models.user import User, active_patients
+from portal.models.overall_status import OverallStatus
+
+# revision identifiers, used by Alembic.
+revision = '55469bdd181f'
+down_revision = '9026af5fe040'
+
+
+def qb_pairs_by_rp():
+    """Lookup and generate a map of RP2->RP3 versions of QBs
+
+    Issue only applies to IRONMAN RPs - ignore others
+    """
+    rp2_id = ResearchProtocol.query.filter_by(name='IRONMAN v2').one().id
+    rp3_id = ResearchProtocol.query.filter_by(name='IRONMAN v3').one().id
+
+    qb_map = dict()  # keyed by the RP2 QB ID
+    for qb in QuestionnaireBank.query.filter_by(research_protocol_id=rp2_id):
+        rp3_name = "IRONMAN_v3" + qb.name[len('IRONMAN'):]
+        rp3_qb = QuestionnaireBank.query.filter_by(name=rp3_name).one()
+        assert (rp3_qb.research_protocol_id == rp3_id)
+        qb_map[qb.id] = rp3_qb.id
+    return qb_map
+
+
+def upgrade():
+    # Tricky task to locate the errors.  One time expensive process,
+    # loop through patients identifying problems.
```

Comment:
>Per email "RE: 170-99-003" 2018-01-17 to truenth-dev


If it's a specific problem, can you give a brief description? Not sure the exact issue after looking at referenced email...

Reply:
Added detailed description to the jira issue.